src/lbaf/IO/lbsVTDataReader.py

```diff
@@ -91,79 +109,158 @@ def _load_vt_file(self, rank_id: int):
+                    f"Ignored phase {curr_phase_id} for rank {rank_id}")
+                continue
+            else:
+                # Desired phase was found
+                phase_id_found = True
+                break
+
+        # Error out if desired phase was not found
+        if not phase_id_found:
+            self.__logger.error(
+                f"Phase {curr_phase_id} not found for rank {rank_id}")
+            sys.excepthook = exc_handler
+            raise SystemExit(1)
+
+        # Proceed with desired phase
+        self.__logger.debug(
+            f"Loading phase {curr_phase_id} for rank {rank_id}")
+
+        # Add communications to the object
+        rank_comm = {}
+        communications = phase.get("communications")
+        if communications:
+            for num, comm in enumerate(communications):
+                # Retrieve communication attributes
+                c_type = comm.get("type")
+                c_to = comm.get("to")
+                c_from = comm.get("from")
+                c_bytes = comm.get("bytes")
+
+                # Supports only SendRecv communication type
+                if c_type == "SendRecv":
+                    # Check whether both are objects
+                    if c_to.get("type") == "object" and c_from.get("type") == "object":
+                        # Create receiver if it does not exist
+                        receiver_obj_id = c_to.get("id")
+                        rank_comm.setdefault(
+                            receiver_obj_id, {"sent": [], "received": []})
+
+                        # Create sender if it does not exist
+                        sender_obj_id = c_from.get("id")
+                        rank_comm.setdefault(
+                            sender_obj_id, {"sent": [], "received": []})
+
+                        # Create communication edges
+                        rank_comm[receiver_obj_id]["received"].append(
+                            {"from": c_from.get("id"),
+                             "bytes": c_bytes})
+                        rank_comm[sender_obj_id]["sent"].append(
+                            {"to": c_to.get("id"), "bytes": c_bytes})
+                        self.__logger.debug(
+                            f"Added communication {num} to phase {curr_phase_id}")
+                        for k, v in comm.items():
+                            self.__logger.debug(f"{k}: {v}")
+
+        # Instantiante rank for current phase
+        phase_rank = Rank(self.__logger, rank_id)
+
+        # Initialize storage for shared blocks information
+        rank_blocks, task_user_defined = {}, {}
+
+        # Iterate over tasks
+        for task in phase.get("tasks", []):
+            # Retrieve required values
+            task_entity = task.get("entity")
+            task_id = task_entity.get("id")
+            task_load = task.get("time")
+            task_user_defined = task.get("user_defined", {})
+            subphases = task.get("subphases")
```

Comment:
I think anything that wasn't explicitly retrieved above should still be retained for output purposes. Maybe add another argument to the object constructor that contains unrecognized parts of the dict?

Reply:
Following our discussion of 05/03 I believe this is is resolved